### DISCUSSION TITLE
High resolution image for face detection

### DISCUSSION THREAD

**Discussion by chokays** [blakeblackshear/frigate#2439] [General]
I’m really impressed with the results I’m getting from Frigate. Great documentation as well.

My setup has 4 8MP/4K Hikvision camera’s (3840x2160). Running Frigate on an Odyssey blue with Intel Celeron J4125, 8GB RAM and a PCI Coral TPU.

I first tried to use the main 4k stream for detection. This worked for 1 camera but the CPU can’t keep up when running all 4 in 4K.

In my current config I use the third stream on my cams set to 1280x720p at 6 fps and the sub stream set to 640x480 for recording and stream. CPU load sits between 20% to 80% which seems okay to me.

Like I said, this works great. But now I want to experiment with face detection. My first thought is to take advantage of the 4K resolution on my camera’s for face detection.  Would this be helpful? And if so, is there anyway to extract an image from the main 4K stream to use for face detection?

I guess I could pull a jpeg from the main 4k stream in Home Assistant when an event is triggered. But looking at the face detection that can be used on top off Frigate, it seems better if the image would be supplied by Frigate and passed on the the MQTT.

Any thoughts on this?

**Comment by blakeblackshear** [comment]
Duplicate of https://github.com/blakeblackshear/frigate/issues/2199
